lib/commons/aria/validate-node-and-attributes.js

```diff
@@ -0,0 +1,89 @@
+/* global aria */
+/**
+ * @description Method that validates a given node against a list of constraints.
+ * @param {HTMLElement} node node to verify attributes against constraints
+ * @param {Array<String|Object>} constraintsArray an array containing TAGNAME or an OBJECT abstraction with conditions and attributes
+ * @return {Boolean} true/ false based on if node passes the constraints expected
+ */
+aria.validateNodeAndAttributes = function validateNodeAndAttributes(
```

Comment:
I think this one would probably be a little easier if instead of passing in an array of constraints, you had passed it one constraint. It simplifies the function quite a bit and it makes it more generic. I think something like a "nodeMatchesDefinition" where we can pass it different things could be useful in other places as well.

I don't think this needs to change now, but it's something to think about.